### ISSUE TITLE
 `Type error: No overload matches this call` w/ `next build` for `./node_modules/@supabase/realtime-js/src/RealtimeClient.ts:294:321

### ISSUE THREAD

**Issue by nbarrow-inspire-labs** [supabase/realtime-js#276]
# Bug report

- [X] I confirm this is a bug with Supabase, not with my own application. 
- [X] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

While running `next build` using Next `^14.0.2`, using `@supabase/supabase-js` (`^2.38.4`), I get the following error: 
```text
./node_modules/@supabase/realtime-js/src/RealtimeClient.ts:294:32
Type error: No overload matches this call.
```
The full traceback stack is below.

```text
info  - Need to disable some ESLint rules? Learn more here: https://nextjs.org/docs/basic-features/eslint#disabling-rules
   Linting and checking validity of types  ..Failed to compile.

./node_modules/@supabase/realtime-js/src/RealtimeClient.ts:294:32
Type error: No overload matches this call.
  Overload 1 of 2, '(input: RequestInfo, init?: RequestInit | undefined): Promise<Response>', gave the following error.
    Argument of type 'string | Request | URL | URL | Request' is not assignable to parameter of type 'RequestInfo'.
      Type 'Request' is not assignable to type 'RequestInfo'.
        Property 'duplex' is missing in type 'Request' but required in type 'import("/Users/nrb/Github/inspire-tms-studio/node_modules/undici-types/fetch").Request'.
  Overload 2 of 2, '(input: URL | RequestInfo, init?: RequestInit | undefined): Promise<Response>', gave the following error.
    Argument of type 'string | Request | URL | URL | Request' is not assignable to parameter of type 'URL | RequestInfo'.
      Type 'Request' is not assignable to type 'URL | RequestInfo'.
        Property 'referrer' is missing in type 'import("/Users/nrb/Github/inspire-tms-studio/node_modules/undici-types/fetch").Request' but required in type 'Request'.

  292 |       _fetch = fetch
  293 |     }
> 294 |     return (...args) => _fetch(...args)
      |                                ^
  295 |   }
  296 |
  297 |   /**
```

## System information

- OS: macOS
- Browser (if applies): Chrome
- Version of supabase-js: `^2.38.4`
- Version of Node.js: `v21.6.0`

## Additional context

This error only started appearing in the last few days or so.

**Comment by nbarrow-inspire-labs** [comment]
Is there any updates on this? It's causing my builds to fail still and I can bypass typescript errors, but then I'm ignoring all errors (not just this one)

**Comment by gokulkrishh** [comment]
@nbarrow-inspire-labs I had a similar issue and it got resolved in `"@supabase/supabase-js": "^2.39.5"`. Try bumping to version `2.39.5` to see if it works for you as your expect it.

**Comment by nbarrow-inspire-labs** [comment]
@gokulkrishh thanks for the heads up, I tried this but it didn't solved the problem. I am still getting, on `^2.39.5`:
```
#0 58.91 ./node_modules/@supabase/realtime-js/src/RealtimeClient.ts:337:32
#0 58.91 Type error: No overload matches this call.
#0 58.91   Overload 1 of 2, '(input: RequestInfo, init?: RequestInit | undefined): Promise<Response>', gave the following error.
#0 58.91     Argument of type 'string | Request | URL | URL | Request' is not assignable to parameter of type 'RequestInfo'.
#0 58.91       Type 'Request' is not assignable to type 'RequestInfo'.
#0 58.91         Property 'duplex' is missing in type 'Request' but required in type 'import("/app/node_modules/undici-types/fetch").Request'.
#0 58.91   Overload 2 of 2, '(input: URL | RequestInfo, init?: RequestInit | undefined): Promise<Response>', gave the following error.
#0 58.91     Argument of type 'string | Request | URL | URL | Request' is not assignable to parameter of type 'URL | RequestInfo'.
#0 58.91       Type 'Request' is not assignable to type 'URL | RequestInfo'.
#0 58.91         Property 'referrer' is missing in type 'import("/app/node_modules/undici-types/fetch").Request' but required in type 'Request'.
#0 58.91 
#0 58.91   335 |       _fetch = fetch
#0 58.91   336 |     }
#0 58.91 > 337 |     return (...args) => _fetch(...args)
#0 58.91       |                                ^
#0 58.91   338 |   }
#0 58.91   339 |
#0 58.91   340 |   /**
```

**Comment by nbarrow-inspire-labs** [comment]
@w3b6x9 when you have a second, would you possibly be able to take a look at this (or ping someone working on this repo)? I would really appreciate any help!! Many thanks in advance!

**Comment by w3b6x9** [comment]
@nbarrow-inspire-labs would love to learn a bit more about your setup. are you using a custom fetch? is the repo you're working on public and if so could you link here?

**Comment by nbarrow-inspire-labs** [comment]
@w3b6x9 sorry for the delay... I finally got around to completely digging this error out. It looks like an old(ish) version of `@types/node` was causing the issue. I scraped through `package-lock.json` and found: 

```
"node_modules/@types/node": {
      "version": "20.9.1",
      "resolved": "https://registry.npmjs.org/@types/node/-/node-20.9.1.tgz",
      "integrity": "sha512-HhmzZh5LSJNS5O8jQKpJ/3ZcrrlG6L70hpGqMIAoM9YVD0YBRNWYsfwcXq8VnSjlNpCpgLzMXdiPo+dxcvSmiA==",
      "dependencies": {
        "undici-types": "~5.26.4"
      }
    },
```

Simply by updating (curiously enough, even though the `undici-types` version did not change) to below fixed the problem:
```
"node_modules/@types/node": {
      "version": "20.11.20",
      "resolved": "https://registry.npmjs.org/@types/node/-/node-20.11.20.tgz",
      "integrity": "sha512-7/rR21OS+fq8IyHTgtLkDK949uzsa6n8BkziAKtPVpugIkO6D+/ooXMvzXxDnZrmtXVfjb1bKQafYpb8s89LOg==",
      "dependencies": {
        "undici-types": "~5.26.4"
      }
    },
```

TLDR; if anyone else encounters this, run a simple `npm update @types/node`.

Sorry to waste your time @w3b6x9 !!

**Comment by w3b6x9** [comment]
@nbarrow-inspire-labs no worries, glad you got it resolved. thanks for posting back here!
